basic/src/test/java/org/jboss/hal/testsuite/test/configuration/webservices/configuration/handlerchain/handler/WebServicesEndpointConfigurationHandlerChainHandlerTestCase.java

```diff
@@ -0,0 +1,206 @@
+package org.jboss.hal.testsuite.test.configuration.webservices.configuration.handlerchain.handler;
+
+import org.apache.commons.lang.RandomStringUtils;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.hal.testsuite.creaper.ResourceVerifier;
+import org.jboss.hal.testsuite.fragment.config.webservices.WebServicesHandlerChainHandlerWizard;
+import org.jboss.hal.testsuite.test.configuration.webservices.WebServicesTestCaseAbstract;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.extras.creaper.core.online.operations.Address;
+import org.wildfly.extras.creaper.core.online.operations.OperationException;
+import org.wildfly.extras.creaper.core.online.operations.Values;
+
+import java.io.IOException;
+import java.util.concurrent.TimeoutException;
+
+@RunWith(Arquillian.class)
+@RunAsClient
+public class WebServicesEndpointConfigurationHandlerChainHandlerTestCase extends WebServicesTestCaseAbstract {
+    private static final String endpointConfigurationName = "TestEndpointConfig_" + RandomStringUtils.randomAlphanumeric(7);
+    private static final Address endpointConfigurationAddress = WEBSERVICES_ADDRESS.and(ENDPOINT_CONFIGURATION, endpointConfigurationName);
+    private static final String preHandlerChainName = "pre_handler_chain_" + RandomStringUtils.randomAlphanumeric(7);
+    private static final Address preHandlerChainAddress = endpointConfigurationAddress.and(PRE_HANDLER_CHAIN, preHandlerChainName);
+    private static final String postHandlerChainName = "post_handler_chain_" + RandomStringUtils.randomAlphanumeric(7);
+    private static final Address postHandlerChainAddress = endpointConfigurationAddress.and(POST_HANDLER_CHAIN, postHandlerChainName);
+
+    @BeforeClass
+    public static void setUp() throws IOException, TimeoutException, InterruptedException {
+        operations.add(endpointConfigurationAddress);
+        administration.reloadIfRequired();
+        operations.add(preHandlerChainAddress);
+        administration.reloadIfRequired();
+        operations.add(postHandlerChainAddress);
+        administration.reloadIfRequired();
```

Comment:
If resources are not dependent on each other, reload after adding isn't needed. One reload after adding `endpointConfigurationAddress` and one after adding both `preHandlerChainAddress` and `postHandlerChainAddress` should be enough.